evm_arithmetization/src/keccak/round_flags.rs

```diff
@@ -56,19 +61,25 @@ pub(crate) fn eval_round_flags_recursively<F: RichField + Extendable<D>, const D
     }
 
     // Flags should circularly increment, or be all zero for padding rows.
+    let current_any_flag =
+        builder.add_many_extension((0..NUM_ROUNDS).map(|i| local_values[reg_step(i)]));
     let next_any_flag =
         builder.add_many_extension((0..NUM_ROUNDS).map(|i| next_values[reg_step(i)]));
+    // Padding row should only start after the last round row.
+    let not_final_step = builder.sub_extension(one, local_values[reg_step(NUM_ROUNDS - 1)]);
+    let padding_constraint = {
+        let tmp = builder.mul_sub_extension(current_any_flag, next_any_flag, current_any_flag);
+        builder.mul_extension(tmp, not_final_step)
```

Comment:
If `not_final_step` is defined the reverse way (`last_round_flag - P::ONES`), then we can call `mul_sub_extension` here and not have the `sub_extension` above.

Reply:
Thanks, missed this. This is a nice trick